src/OnlineRetailPortal.Mock/MockProductStore.cs

```diff
@@ -46,7 +46,7 @@ public async Task<AddProductStoreResponse> AddProductAsync(AddProductStoreReques
 
         public async Task<GetProductStoreResponse> GetProductAsync(string productId)
         {
-            Product response = productList.Where(x => x.Id == productId).First();
+            Product response = productList.Where(x => x.Id == productId).FirstOrDefault();
```

Comment:
is it checked somewhere that productId is anyhow going to be present in the productList? what if its not present

Reply:
productId is the primary key for products, it can not be null...?